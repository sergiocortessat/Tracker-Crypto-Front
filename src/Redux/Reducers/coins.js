const initialState = {
  coins: [],
};

const coinReducer = (state = initialState, action) => {
  switch (action.type) {
    case 'ALL_COINS':
      return {
        ...state,
        coins: action.payload,
      };
    default:
      return state;
  }
};

export default coinReducer;
