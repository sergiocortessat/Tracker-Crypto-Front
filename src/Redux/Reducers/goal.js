const initialState = {
  goals: [],
};

const goalReducer = (state = initialState, action) => {
  switch (action.type) {
    case 'ALL_GOALS':
      return {
        ...state,
        goal: action.payload,
      };
    default:
      return state;
  }
};

export default goalReducer;
