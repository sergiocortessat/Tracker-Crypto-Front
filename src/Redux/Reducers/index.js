import { combineReducers } from 'redux';
import Coins from './coins';
import User from './user';

const reducers = combineReducers({
  coins: Coins,
  user: User,
});

export default reducers;
