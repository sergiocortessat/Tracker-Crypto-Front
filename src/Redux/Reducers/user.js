const initialState = {
  user: [],
};

const userReducer = (state = initialState, action) => {
  switch (action.type) {
    case 'CURRENT_USER':
      return {
        ...state,
        user: action.payload,
      };
    default:
      return state;
  }
};

export default userReducer;
