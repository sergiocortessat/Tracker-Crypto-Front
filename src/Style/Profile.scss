$grey: #67899c;
$green: #addc91;
$blue: #62b5e5;

.profile {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 39rem;
}

.a-box {
  display: inline-block;
  width: 240px;
  text-align: center;
  margin-bottom: 140px;
}

.img-container {
  height: 230px;
  width: 200px;
  overflow: hidden;
  border-radius: 0 0 20px 20px;
  display: inline-block;
}

.img-container img {
  transform: skew(0deg, -13deg);
  height: 250px;
  margin: -35px 0 0 -70px;
}

.inner-skew {
  display: inline-block;
  border-radius: 20px;
  overflow: hidden;
  padding: 0;
  transform: skew(0deg, 13deg);
  font-size: 0;
  margin: 30px 0 0 0;
  background: #c8c2c2;
  height: 250px;
  width: 200px;
}

.text-container {
  box-shadow: 0 0 10px 0 rgba(0, 0, 0, 0.2);
  padding: 120px 20px 20px 20px;
  border-radius: 20px;
  background: #fff;
  margin: -120px 0 0 0;
  line-height: 19px;
  font-size: 14px;
  color: $grey;
}

.text-container h3 {
  margin: 20px 0 10px 0;
  color: $grey;
  font-size: 18px;
}

.created-at {
  margin-top: 10px;
}
