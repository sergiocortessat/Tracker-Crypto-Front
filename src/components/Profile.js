/* eslint-disable no-unused-vars */
import React from 'react';
import { useSelector } from 'react-redux';
import '../Style/Profile.scss';

const Profile = () => {
  const { user } = useSelector((state) => state.user);
  return (
  // <div>
  //   <p>
  //     User Name:
  //   </p>
  //   <p>
  //     {user.name}
  //   </p>
  //   <p> User Email: </p>
  //   <p>{user.email}</p>
  //   <img src={user.picture} alt="avatar" />
  //   <p>
  //     Account created at:
  //   </p>
  //   <p>
  //     {user.created_at}
  //   </p>
  // </div>
    <div className="profile">
      <div className="a-box">
        <div className="img-container">
          <div className="img-inner">
            <div className="inner-skew">
              <img
                alt="profile"
                src={user.picture ? user.picture : 'https://www.kindpng.com/picc/m/252-2524695_dummy-profile-image-jpg-hd-png-download.png'}
              />
            </div>
          </div>
        </div>
        <div className="text-container">
          <h3>{user.name}</h3>
          <div>
            {user.email}
          </div>
          <div className="created-at">
            This accont was created on:
            {' '}
            {user.created_at}
          </div>
        </div>
      </div>
    </div>
  );
};

export default Profile;
