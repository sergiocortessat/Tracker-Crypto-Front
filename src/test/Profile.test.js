import { Provider } from 'react-redux';
import { createStore } from 'redux';
import renderer from 'react-test-renderer';
import Reducers from '../Redux/Reducers';
import Profile from '../components/Profile';

const store = createStore(Reducers);

it('renders correctly', () => {
  const addMeasurement = renderer
    .create(
      <Provider store={store}>
        <Profile />
      </Provider>,
    )
    .toJSON();
  expect(addMeasurement).toMatchSnapshot();
});
