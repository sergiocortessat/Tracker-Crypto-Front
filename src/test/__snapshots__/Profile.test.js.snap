// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="profile"
>
  <div
    className="a-box"
  >
    <div
      className="img-container"
    >
      <div
        className="img-inner"
      >
        <div
          className="inner-skew"
        >
          <img
            alt="profile"
            src="https://www.kindpng.com/picc/m/252-2524695_dummy-profile-image-jpg-hd-png-download.png"
          />
        </div>
      </div>
    </div>
    <div
      className="text-container"
    >
      <h3 />
      <div />
      <div
        className="created-at"
      >
        This accont was created on:
         
      </div>
    </div>
  </div>
</div>
`;
